src/main/webapp/robot.js

```diff
@@ -112,3 +114,40 @@ function hideStackTrace(id) {
     document.getElementById(id + "-showlink").style.display = "";
     document.getElementById(id + "-hidelink").style.display = "none";
 }
+
+document.addEventListener("DOMContentLoaded", function(){
+    var element = document.getElementById("robot-fromurl-id");
+    if (element) {
+        var url = element.getAttribute("data-url");
+
+        ["zoomToChanges", "failedOnly", "criticalOnly"].forEach(function(id) {
+            if (document.getElementById(id))
+                document.getElementById(id).addEventListener("click", redrawPassFailGraph);
+        });
+
+        if (document.getElementById("maxBuildsToShow")) {
+            document.getElementById("maxBuildsToShow").addEventListener("change", redrawPassFailGraph);
+            if (url == "")
+                document.getElementById("maxBuildsToShow").addEventListener("change", redrawDurationGraph);
+        }
+        initPassFailGraph(url);
+        if (url == "")
+            initDurationGraph(url);
+    }
+});
+
+document.addEventListener("DOMContentLoaded", function(){
```

Comment:
```suggestion

```

Reply:
I don't think you can define two different functions to `DOMContentLoaded` without overwriting the former with the latter. These should probably be combined together as a single startup

Reply:
Worked on my machine, but I'll refactor this. It probably makes more sense to have them combined anyway.